sympy/functions/combinatorial/factorials.py

```diff
@@ -1167,19 +1158,22 @@ class multinomial(CombinatorialFunction):
     .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
 
     """
-    def __new__(cls, *args):
+    @classmethod
+    def eval(cls, *args):
         if not args:
             raise ValueError("A miniumum of 1 arg required")
-        args = [i for i in ordered(map(sympify, args)) if not i.is_zero]
+
         if len(args) < 2:
             # all args might have been zero or there might be only
             # one arg; in either case the simple result is returned
             return S.One
-        return super(multinomial, cls).__new__(
-            cls, *args)
 
-    @classmethod
-    def eval(cls, *args):
+        if any(i.is_zero for i in ordered(map(sympify, args))):
+            args = [i for i in ordered(map(sympify, args)) if not i.is_zero]
```

Comment:
The `if` is not needed. The comprehension does two things: filters out zeros and orders the args.

Reply:
Ohhh that, remember when we had that `__new__` method for removing zeros from multinomial ... such that we'd get
```python3
>>> multinomial(n, k, z, z, z)
multinomial(n, k)
```
that's what this loop triggers
```python3
        if any(i.is_zero for i in ordered(map(sympify, args))):
            args = [i for i in ordered(map(sympify, args)) if not i.is_zero]
            if not args:
                args = [S.Zero]
            return multinomial(*args)
```
any other ideas on how to do this without using `__new__` ?